src/tests/ftest/util/nvme_utils.py

```diff
@@ -38,6 +40,74 @@ def get_device_ids(test, dmg, servers):
     return devices
 
 
+def set_device_faulty(test, dmg, server, uuid, pool=None, **kwargs):
+    """Set the device faulty and optionally wait for rebuild to complete.
+
+    Args:
+        test (Test): avocado test class
+        dmg (DmgCommand): a DmgCommand class instance
+        server (NodeSet): host on which to issue the dmg storage set nvme-faulty
+        uuid (str): the device UUID
+        pool (TestPool, optional): pool used to wait for rebuild to start/complete if specified.
+            Defaults to None.
+        kwargs (dict, optional): named arguments to pass to the DmgCommand.storage_set_faulty.
+
+    Returns:
+        dict: the json response from the dmg storage set-faulty command.
+
+    """
+    dmg.hostlist = server
+    kwargs['uuid'] = uuid
+    # Use this when DmgCommand.storage_set_faulty returns json output natively
+    # response = get_dmg_response(dmg.storage_set_faulty, **kwargs)
+
+    # Remove this when DmgCommand.storage_set_faulty returns json output natively
+    dmg_json = dmg.json.value
+    dmg.json.value = True
+    try:
+        res = dmg.storage_set_faulty(**kwargs)
+    except CommandFailure as error:
+        test.fail('dmg.storage_set_faulty({}) failed: {}'.format(dict_to_str(kwargs), error))
+    finally:
+        dmg.json.value = dmg_json
+    response = get_json_response(test, json.loads(res.stdout_text), None,
+                                 'dmg.storage_set_faulty({})'.format(dict_to_str(kwargs)))
```

Comment:
Now that https://github.com/daos-stack/daos/commit/e4150a5df1c1bbefccc4727d8e2fcb80f51758df has landed please make the following change:
```suggestion
    response = get_dmg_response(dmg.storage_set_faulty, **kwargs)
```

Reply:
With these changes the PR should run:
- test_dmg_storage_query_device_state
- test_disk_failure_w_rf
- test_disk_fault_to_normal
- NvmeEnospace
- test_nvme_fault
- OSAOfflineDrain
- OSAOfflineReintegration
- EcodAggregationOff
- test_ec_multiple_rank_failure
- test_ec_multiple_targets_on_same_rank
- test_ec_multiple_targets_on_diff_ranks
- test_ec_single_target_rank_failure
- EcodAggregationOffRebuild
- EcodOfflineRebuild
- test_ec_online_rebuild
- test_ec_degrade
- EcodServerRestart
- test_ec_ior_hard_online_rebuild

Reply:
Please see the new commit.

Reply:
done